### PR TITLE
config: Add Eden chain context and genesis heights

### PR DESCRIPTION
**Pull request by ptrus**

### CODE REVIEW

File: config/history.go [hunk 1]
```diff
@@ -80,21 +80,20 @@ func (h *History) RecordForRuntimeRound(runtime common.Runtime, round uint64) (*
...
-		},
-	}
-}
-
 var DefaultChains = map[common.ChainName]*History{
 	common.ChainNameMainnet: {
 		Records: []*Record{
+			{
+				// https://github.com/oasisprotocol/mainnet-artifacts/releases/tag/2023-11-29
+				ArchiveName:   "eden",
```
Comment (mitjat) [inline config/history.go:88]: Just FYI, this archive name is referenced [here](https://github.com/oasisprotocol/nexus/blob/6b2fe2e4d580e7f9be194081684c19dcf7f1a6bf/storage/oasis/nodeapi/history/history.go#L50-L50) to tell nexus what API to use to talk to the node. (The 23.0.x API is called "eden" internally.) Should be preconfigured correctly, but hasn't been tested, obviously.

**Review by mitjat — APPROVED** [review]
:+1: I believe that should be all that's needed. Thank you!!

For reference, here are the changes I made for enabling 23.0.x on testnet:
- [nexus](https://github.com/oasisprotocol/nexus/commit/3fa36595db47926b69af05e50009290edc3a657e) (only the config/history changes are relevant now)
- [ops](https://github.com/oasislabs/private-ops/pull/2772/files)

File: config/history.go [hunk 2]
```diff
@@ -80,21 +80,20 @@ func (h *History) RecordForRuntimeRound(runtime common.Runtime, round uint64) (*
...
+				GenesisHeight: 16817956,
+				RuntimeStartRounds: map[common.Runtime]uint64{
+					common.RuntimeCipher:   44054,
+					common.RuntimeEmerald:  7875129,
+					common.RuntimeSapphire: 1357486,
+				},
+				ChainContext: "bb3d748def55bdfb797a2ac53ee6ee141e54cd2ab2dc2375f4a0703a178e6e55",
+			},
```
Comment (ptrus) [inline config/history.go:96]: Someone doublecheck these please ^

**Comment by ptrus** [comment]
Build and pushed `oasislabs/oasis-indexer:eden-ptrus ` bult on latest commit in this PR.

File: config/history.go (hunk 2 above)
Comment (mitjat) [reply config/history.go:96]: :+1: Double-checked the chain context against the release page, and the heights against the contents of `genesis.json` that's posted there.

**Review by mitjat — APPROVED** [review]
Thank you! This is running in staging _and_ prod now, so that's another confirmation we're good :+1:

**Pull request merged**